src/lib/components/forms/form/form-components/multi-select/multi-select.component.spec.ts

```diff
@@ -1,24 +1,202 @@
+            declarations: [MultiSelectComponent, MockTooltipDirective, SelectSortPipe]
+        });
 
-    beforeEach(() => {
         fixture = TestBed.createComponent(MultiSelectComponent);
         component = fixture.componentInstance;
+        loader = TestbedHarnessEnvironment.loader(fixture);
+
         fixture.detectChanges();
+
+        select = await loader.getHarness(MatSelectHarness);
     });
 
-    it('should create', () => {
+    it('should create instances', async () => {
         expect(component).toBeTruthy();
+        expect(select).toBeTruthy();
+    });
+
+    it('should check whether the select is in multi-selection mode', async () => {
+        expect(await select.isMultiple()).toBe(true);
+    });
+
+    it('should disable the select when #isDisabled is set', async () => {
+        expect(await select.isDisabled()).toBe(false);
+
+        component.isDisabled = true;
+        fixture.detectChanges();
+
+        expect(await select.isDisabled()).toBe(true);
+    });
+
+    it('should set required validation when #isRequired is set', async () => {
+        expect(await select.isRequired()).toBe(false);
+
+        component.isRequired = true;
+        fixture.detectChanges();
+
+        expect(await select.isRequired()).toBe(true);
+    });
+
+    it('should set #name as label of the select', async () => {
+        const formField: MatFormFieldHarness = await loader.getHarness(MatFormFieldHarness);
+        expect(await formField.getLabel()).toBe('');
+
+        component.name = 'Ma Label';
+        fixture.detectChanges();
+
+        expect(await formField.getLabel()).toBe(component.name);
+    });
+
+    it('should call the callback #registerOnChange whenever the value of an option is changed by the user', async () => {
+        const spy: jasmine.Spy = jasmine.createSpy('onChangeCallback');
+        component.registerOnChange(spy);
+        component.listBoxValues = multiSelectOptions;
+
+        fixture.detectChanges();
+
+        await select.clickOptions({
+            text: multiSelectOption1.caption.toString(),
+            isSelected: true
+        });
+
+        expect(spy).toHaveBeenCalledOnceWith(multiSelectOption1.value, true);
+    });
+
+    it('should call the callback #registerOnTouched whenever the select was blurred', async () => {
+        const onTouchedCallback: jasmine.Spy = jasmine.createSpy('onTouched');
+        component.registerOnTouched(onTouchedCallback);
+
+        await select.blur();
+
+        expect(onTouchedCallback).toHaveBeenCalled();
+    });
+
+    it('should set the value from ngModel as from #writeValue', async () => {
+        component.listBoxValues = multiSelectOptions;
+        component.writeValue([multiSelectOption1.value, multiSelectOption2.value]);
+
+        fixture.detectChanges();
+
+        expect(await select.getValueText()).toBe([multiSelectOption1.value, multiSelectOption2.value].toString());
```

Comment:
Ich denke es wird dann die `caption` der option, nicht der `value` angezeigt.
```suggestion
        expect(await select.getValueText()).toBe([multiSelectOption1.caption, multiSelectOption2.caption].toString());
```
Ansonsten gehst du über der liste der options und schaust ob die richtigen selektiert sind.